core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -212,7 +212,7 @@ final class KafkaMetadataLog private (
   }
 
   override def flush(): Unit = {
-    log.flush()
+    log.flushUpToAndExcludingLogEndOffset()
```

Comment:
It seems that the Raft log could have the same issue during a clean shutdown. The easiest fix is to flush it inclusively during close.

Reply:
Used `log.flush(true)`. Are we concerned about performance by this?